### DISCUSSION TITLE
Different simulation results between ngspice and  Xyce

### DISCUSSION THREAD

**Discussion by dbm-174** [IHP-GmbH/IHP-Open-PDK#294] [Q&A] [answered]
Hi all, 
I tried to do a small simulation with the npn13G2, where I plot Ic vs Vce. In my simulation ngspice (blue) and Xyce (red) show different results. 
![simulation_output](https://github.com/user-attachments/assets/9ab43755-59c9-413a-a6a9-2cc6c91b45e7)

Does anyone know what could be the reason for this behavior?

Thank you very much in advance

**ngspice netlist**

*Qucs 24.4.1  /home/bastian/.qucs/lna_prj/dc_hbt_13g2.sch
.INCLUDE "/foss/tools/qucs-s/current/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc"

.SUBCKT IHP_PDK_nonlinear_components_npn13G2  gnd c b e bn Nx=1  
X1 c b e bn npn13G2 Nx={Nx} 
.ENDS
  
*.LIB cornerHBT.lib hbt_typ
.LIB /foss/pdks/IHP-Open-PDK/ihp-sg13g2/libs.tech/xyce/models/cornerHBT.lib  hbt_typ

Vbe _net0 0 DC 1
VPrIc _net1 _net2 DC 0
VPrIb _net0 _net3 DC 0
Vce _net1 0 DC 1.5
Xnpn13G1 0  _net2 _net3 0 gnd IHP_PDK_nonlinear_components_npn13G2 Nx=1

.control

op
print i(VPrIb) i(VPrIc)   > spice4qucs.dc1.ngspice.dc.print
destroy all
reset

dc vce 0 2 0.00666667
write spice4qucs.sw1.plot i(VPrIb) i(VPrIc)
destroy all
reset

exit
.endc
.END

**Xyce netlist:**

*Qucs 24.4.1  /home/bastian/.qucs/lna_prj/dc_hbt_13g2.sch

.SUBCKT IHP_PDK_nonlinear_components_npn13G2  gnd c b e bn Nx=1  
X1 c b e bn npn13G2 Nx={Nx} 
.ENDS
  
*.LIB cornerHBT.lib hbt_typ
.LIB /foss/pdks/IHP-Open-PDK/ihp-sg13g2/libs.tech/xyce/models/cornerHBT.lib  hbt_typ

Vbe _net0 0 DC 1
VPrIc _net1 _net2 DC 0
VPrIb _net0 _net3 DC 0
Vce _net1 0 DC 1.5
Xnpn13G1 0  _net2 _net3 0 gnd IHP_PDK_nonlinear_components_npn13G2 Nx=1

.dc vce 0 2 0.00666667
.PRINT  dc format=raw file=spice4qucs.dc.plot I(VPrIb) I(VPrIc) 
.END

**Comment by dwarning** [comment]
It's not the root cause, but you should use ngspice lib in ngspice simulation. And first subckt node for ground is obsolete. In my opinion ngspice result is correct.

**Comment by dwarning** [comment]
The xyce lib is wrong. Thermal node t in the subckt is floating and selfheating is on. Please look for the messages by Xyce in the beginning of simulation (hope it is shown in qucs-s).

**Comment by KrzysztofHerman** [comment]
@dbm-174 

What I suppose is that the last IIC-OSIC-TOOL docker image does not include the update we have made recently in PR #283 related to the PR #261 where models and symbols for Xyce and ngspice where introduced. 

Here I provide you the netlists generated by xschem for ngspice and Xyce. 
There i no difference in the simulation results, 

Ngspice
```
**.subckt dc_hbt_13g2
Vce net3 GND 0.0
I0 GND net1 0.0
Vc net3 net2 0
.save i(vc)
XQ1 net2 net1 GND GND npn13G2 Nx=1
**** begin user architecture code
.lib cornerHBT.lib hbt_typ
.param temp=27
.control
save all
dc Vce 0 1.2 0.01 I0 0 5u 0.1u
write dc_hbt_13G2.raw
.endc
**** end user architecture code
**.ends
.GLOBAL GND
.end

```

Xyce 
```
**.subckt dc_hbt_13g2
Vce net3 GND 0.0
I0 GND net1 0.0
Vc net3 net2 0
XQ1 net2 net1 GND GND npn13G2 Nx=1
**** begin user architecture code
.preprocess replaceground true
.option temp=27
.step I0 0 5u 0.1u
.dc Vce 0 1.2 0.01
.print dc format=raw file=dc_hbt_13G2.raw I(Vc)
.lib  your_location/IHP-Open-PDK/ihp-sg13g2/libs.tech/xyce/models/cornerHBT.lib hbt_typ
**** end user architecture code
**.ends
.GLOBAL GND
.end
```

![image](https://github.com/user-attachments/assets/c02a8b2d-91a7-4aa3-b58c-ec62cfcd5e95)

@dwarning could you please review it ?

**Comment by dbm-174** [comment, marked as answer]
Many thanks for the fast and very helpful replies.
I did not use the docker image. I used  the main branch of the IHP repository and switched to the most recent dev. Now the output is the same (ngspice=blue stars).

Thanks again and sorry to bother!

![simulation_output_new](https://github.com/user-attachments/assets/c8f64669-1796-4672-a48a-6663a15eea0b)
